log/sublogger.go

```diff
@@ -53,15 +52,16 @@ func (sl *SubLogger) getFields() *logFields {
 		return nil
 	}
 
+	fields := logFieldsPool.Get().(*logFields) // nolint:forcetypeassert // Not necessary from a pool
+
 	sl.mtx.RLock()
 	defer sl.mtx.RUnlock()
-	return &logFields{
-		info:   sl.levels.Info,
-		warn:   sl.levels.Warn,
-		debug:  sl.levels.Debug,
-		error:  sl.levels.Error,
-		name:   sl.name,
-		output: sl.output,
-		logger: logger,
-	}
+	fields.info = sl.levels.Info
+	fields.warn = sl.levels.Warn
+	fields.debug = sl.levels.Debug
+	fields.error = sl.levels.Error
+	fields.name = sl.name
+	fields.output = sl.output
+	fields.logger = logger
```

Comment:
What if you change your pool New definition to
`	logFieldsPool = &sync.Pool{New: func() interface{} { return &logFields{logger: logger} }}`

Then you don't need to assign the logger every time, even if its not new

Reply:
I'm resolving this due to issues raised around races and outdated loggers. Doesn't need to be addressed in this PR